### PR TITLE
Wrap the parameters of the Ephemeris functions into parameter classes

### PR DESCRIPTION
**Pull request by pleroy**
No semantic changes at this point.  In particular, `max_steps` is still hard-wired to 1000.  Partially addresses #911.

### CODE REVIEW

File: ksp_plugin/flight_plan.cpp [hunk 1]
```diff
@@ -124,8 +123,8 @@ bool FlightPlan::SetFinalTime(Instant const& final_time) {
 void FlightPlan::SetTolerances(
```
Comment (eggrobin) [inline ksp_plugin/flight_plan.cpp]: Make this `SetAdaptiveParameters` (there's no reason to keep the integrator the same if we recompute everything).

File: physics/ephemeris.hpp [hunk 2]
```diff
@@ -42,15 +42,52 @@ class Ephemeris {
...
+
+    template<typename T>
+    void WriteToMessage(not_null<T*> const t) const;
+
+    template<typename T>
+    static AdaptiveStepParameters ReadFromMessage(T const& t);
+
+   private:
+    // This will refer to a static object returned by a factory.
+    AdaptiveStepSizeIntegrator<NewtonianMotionEquation> const& integrator_;
```
Comment (eggrobin) [inline physics/ephemeris.hpp]: This makes the class non-assignable, which is a bit annoying for something that's just a parameter container.We tend to use pointers in those cases (the exception to our reference conventions is that we may take the address of a reference to a static object, this is also somewhat easier to enforce).
Make it a `not_null<AdaptiveStepSizeIntegrator<NewtonianMotionEquation>>`.

File: physics/ephemeris.hpp [hunk 3]
```diff
@@ -42,15 +42,52 @@ class Ephemeris {
...
+   public:
+    // The |length_| and |speed_integration_tolerance|s are used to compute the
+    // |tolerance_to_error_ratio| for step size control.
+    AdaptiveStepParameters(
+        AdaptiveStepSizeIntegrator<NewtonianMotionEquation> const& integrator,
+        Length const& length_integration_tolerance,
+        Speed const& speed_integration_tolerance);
+
+    void SetTolerances(Length const& length_integration_tolerance,
+                       Speed const& speed_integration_tolerance);
```
Comment (eggrobin) [inline physics/ephemeris.hpp]: Once this is assignable we don't need this function (1 use) anymore, remove.

File: ksp_plugin/flight_plan.cpp (hunk 1 above)
Comment (pleroy) [inline ksp_plugin/flight_plan.cpp]: Done.

File: physics/ephemeris.hpp (hunk 2 above)
Comment (pleroy) [inline physics/ephemeris.hpp]: Done.

**Pull request merged**